samples/drivers/spi_flash/src/main.c

```diff
@@ -9,6 +9,18 @@
 #include <device.h>
 #include <stdio.h>
 
+#if (CONFIG_SPI_FLASH_W25QXXDV - 0)
```

Comment:
The subtraction with 0 is for the case when `CONFIG_xx` is not defined? It's the first time I see it.

Reply:
Yes.  This is a standard technique when you want to support the case of defining a flag explicitly to zero, or to an empty token sequence, or leaving it undefined, with all three options producing the same behavior (the condition is false when it expands to `( - 0)`.  It's more flexible than assuming the presence of a definition means "true", as with `#ifdef`.  I don't remember where I saw it first.

Some of these flags may not be defined at all, depending on other build options.  It might not be necessary for `CONFIG_*`.  I primarily chose it because mixing `#ifdef` with `#elif defined` is ugly, but also because I'm more sure it'll do what I need here in all cases.